### PR TITLE
[AppConfig] BREAKING CHANGE: `az appconfig feature`: Disallow colon character in feature name

### PR DESCRIPTION
**Pull request by albertofori**
**Related command**

`az appconfig feature`
`az appconfig kv import`

**Description**

Related to [Azure App Configuration Issue 733](https://github.com/Azure/AppConfiguration/issues/733). Feature name validations are updated to ensure that the feature names do not contain the colon (`:`)
**Testing Guide**

**History Notes**

[AppConfig] BREAKING CHANGE: `az appconfig feature`: Update feature name validation to disallow the colon character
[AppConfig] BREAKING CHANGE: `az appconfig kv import`: Update feature name validation. Invalid feature flags will be skipped during import

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AppConfig

**Comment by albertofori** [comment]
@avanigupta Could you please take a look at this?

File: src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py [hunk 1]
```diff
@@ -119,15 +120,21 @@ def validate_import_key(key):
...
 def validate_import_feature(feature):
-    if feature:
-        if '%' in feature:
-            logger.warning("Ignoring invalid feature '%s'. Feature name cannot contain the '%%' character.", feature)
-            return False
-    else:
-        logger.warning("Ignoring invalid feature ''. Feature name cannot be empty.")
+    try:
+        validate_feature_flag_name(feature)
+        return True
```
Comment (avanigupta) [inline src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py:125]: >return True [](http://example.com/codeflow?start=8&length=11)

Can we move the `return true` statement outside try/catch flow? It just makes the code easier to read to see a default return value if everything succeeds.

```
try {...}
except {...}
return True
```

**Review by avanigupta — DISMISSED** [review]

**Review by avanigupta — DISMISSED** [review]

**Comment by zhoxing-ms** [comment]
@albertofori Could you please fix the CLI style issue?

**Comment by zhoxing-ms** [comment]
Due to the inclusion of breaking change in this PR, we usually recommend releasing it in the next breaking change release (Build Sprint on **05-23**) , so as to reduce the impact of breaking change on customers. @albertofori May I ask if this release time is an acceptable date?

**Comment by zhoxing-ms** [comment]
> May I ask if this release time is an acceptable date?

@albertofori OK, thanks for fixing this issue. What about this question?

**Comment by albertofori** [comment]
> > Due to the inclusion of breaking change in this PR, we usually recommend releasing it in the next breaking change release (Build Sprint on 05-23) , so as to reduce the impact of breaking change on customers. @albertofori May I ask if this release time is an acceptable date?
> 
> @albertofori OK, thanks for fixing this CLI issue. What about this question?

Hi @zhoxing-ms Yes please, the release time works for me. Thanks!

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**